R/model.R

```diff
@@ -959,6 +959,178 @@ sample_method <- function(data = NULL,
+#'
+#' @template seealso-docs
+#' @inherit cmdstan_model examples
+#'
+NULL
+mpi_sample_method <- function(data = NULL,
+                          mpi_cmd = "mpiexec",
+                          mpi_args = NULL,
+                          seed = NULL,
+                          refresh = NULL,
+                          init = NULL,
+                          save_latent_dynamics = FALSE,
+                          output_dir = NULL,
+                          chains = 1,
+                          chain_ids = seq_len(chains),
+                          iter_warmup = NULL,
+                          iter_sampling = NULL,
+                          save_warmup = FALSE,
+                          thin = NULL,
+                          max_treedepth = NULL,
+                          adapt_engaged = TRUE,
+                          adapt_delta = NULL,
+                          step_size = NULL,
+                          metric = NULL,
+                          metric_file = NULL,
+                          inv_metric = NULL,
+                          init_buffer = NULL,
+                          term_buffer = NULL,
+                          window = NULL,
+                          fixed_param = FALSE,
+                          sig_figs = NULL,
+                          validate_csv = TRUE,
+                          show_messages = TRUE) {
+  parallel_chains <- 1
+  if (fixed_param) {
+    chains <- 1
+    save_warmup <- FALSE
+  }
+
+  checkmate::assert_integerish(chains, lower = 1, len = 1)
+  checkmate::assert_integerish(parallel_chains, lower = 1, null.ok = TRUE)
+  checkmate::assert_integerish(chain_ids, lower = 1, len = chains, unique = TRUE, null.ok = FALSE)
+  sample_args <- SampleArgs$new(
+    iter_warmup = iter_warmup,
+    iter_sampling = iter_sampling,
+    save_warmup = save_warmup,
+    thin = thin,
+    max_treedepth = max_treedepth,
+    adapt_engaged = adapt_engaged,
+    adapt_delta = adapt_delta,
+    step_size = step_size,
+    metric = metric,
+    metric_file = metric_file,
+    inv_metric = inv_metric,
+    init_buffer = init_buffer,
+    term_buffer = term_buffer,
+    window = window,
+    fixed_param = fixed_param
+  )
+  cmdstan_args <- CmdStanArgs$new(
+    method_args = sample_args,
+    model_name = strip_ext(basename(self$exe_file())),
+    exe_file = self$exe_file(),
+    proc_ids = chain_ids,
+    data_file = process_data(data),
+    save_latent_dynamics = save_latent_dynamics,
+    seed = seed,
+    init = init,
+    refresh = refresh,
+    output_dir = output_dir,
+    validate_csv = validate_csv,
+    sig_figs = sig_figs
+  )
+  cmdstan_procs <- CmdStanMCMCProcs$new(
+    num_procs = chains,
+    parallel_procs = parallel_chains,
+    show_stderr_messages = show_messages
+  )
+  runset <- CmdStanRun$new(args = cmdstan_args, procs = cmdstan_procs)
+  runset$run_cmdstan_mpi(mpi_cmd, mpi_args)
+  CmdStanMCMC$new(runset)
+}
+CmdStanModel$set("public", name = "mpi_sample", value = mpi_sample_method)
```

Comment:
I'm ok with the name `mpi_sample`, but how do you feel about `sample_mpi`? two reasons. 1) it comes up in autocomplete when people are looking for sample, 2) it's really the same sample method with some extra mpi commands so it's also good to emphasize the "sample" before the "mpi".  But definitely not a huge deal.

Reply:
I agree with both reasons. Good call.